crates/integrations/datafusion/src/table/mod.rs

```diff
@@ -140,8 +152,15 @@ impl TableProvider for IcebergTableProvider {
         filters: &[Expr],
         _limit: Option<usize>,
     ) -> DFResult<Arc<dyn ExecutionPlan>> {
+        // Refresh table if catalog is available
+        let table = if let Some(catalog) = &self.catalog {
+            catalog.load_table(self.table.identifier()).await.map_err(to_datafusion_error)?
+        } else {
+            self.table.clone()
+        };
+
```

Comment:
This makes me wonder that if storing table directly in the `IcebergTableProvider` is correct... We could get a stale table if the provider doesn't have a catalog table. 

Iceberg-java has a [`refresh()`](https://github.com/apache/iceberg/blob/7477f8bfbcf9cded7129bbf8ca7e91bcaa81c1ae/api/src/main/java/org/apache/iceberg/Table.java#L42) interface which uses `TableOperation` to refresh metadata. In iceberg-rs we don't have `TableOperation` and has to rely on catalog to refresh

Reply:
I have a PR to add that functionality (including the refresh): https://github.com/apache/iceberg-rust/pull/1297